### PR TITLE
[EHT-592] Feature/snapshot sync

### PR DESCRIPTION
**Pull request by kowshikvajrala**

### CODE REVIEW

File: roles/snapshot_sync/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  ansible.builtin.file:
+    path: "{{ item }}"
+    state: directory
+    owner: "{{ service_identifier }}"
+    group: "{{ service_identifier }}"
+    mode: "0755"
+  loop:
+    - "{{ snapshot_sync_authorities_data_dir }}"
+    - "{{ snapshot_sync_authorities_data_dir }}/live"
+    - "{{ snapshot_sync_consensus_data_dir }}"
```
Comment (aditya-manit) [inline roles/snapshot_sync/tasks/main.yml:34]: do we need to create this explicitly ?

File: roles/snapshot_sync/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+- name: Download sui formal snapshot using sui-tool
+  ansible.builtin.command:
+    cmd: >
+      {{ snapshot_sync_bin_dir }}/{{ snapshot_sync_tool_default_binary_name }} download-formal-snapshot --latest
+      --genesis {{ snapshot_sync_genesis_file_path }}
+      --network {{ env }}
+      --path {{ snapshot_sync_authorities_data_dir }}/live
+      --num-parallel-downloads 50
```
Comment (aditya-manit) [inline roles/snapshot_sync/tasks/main.yml:43]: does it have to be "{{ snapshot_sync_authorities_data_dir }}/live" or just "{{ snapshot_sync_authorities_data_dir }}"

**Review by aditya-manit — CHANGES_REQUESTED** [review]
Config needs to be updated: https://www.notion.so/encapsulate/Sui-Ansible-15914f565a5080c8bf27da5f08166bc6?pvs=4#16c14f565a5080a9b73df7af35624a53

**Comment by aditya-manit** [comment]
Also lets add group here https://github.com/encapsulate-xyz/sui-ansible/blob/main/tasks/create_user_and_directories.yml#L4

File: roles/snapshot_sync/tasks/main.yml (hunk 2 above)
Comment (kowshikvajrala) [reply roles/snapshot_sync/tasks/main.yml:43]: Yes in their documentation they mentioned to resotre it to that directory

Comment (kowshikvajrala) [reply roles/snapshot_sync/tasks/main.yml:43]: https://docs.sui.io/guides/operator/snapshots#:~:text=When%20you%20restore,the%20path%20location.

Comment (aditya-manit) [reply roles/snapshot_sync/tasks/main.yml:43]: I think we still may just want to use /opt/sui/db/authorities_db and then have to rename the directory
Screenshot 2024-12-30 at 4 02 27 PM

**Comment by aditya-manit** [comment]
Please update https://github.com/encapsulate-xyz/sui-ansible/blob/main/group_vars/testnet.yml#L6 too and lets merge

**Review by aditya-manit — APPROVED** [review]

**Pull request merged**